testsuite/features/step_definitions/command_steps.rb

```diff
@@ -580,12 +580,12 @@
     raise ScriptError, 'File injection failed' unless success
   end
   server.run('chmod +x /etc/ipmi/fake_ipmi_host.sh', verbose: true, check_errors: true)
-  server.run('ipmi_sim -n < /dev/null > /dev/null &', verbose: true, check_errors: true)
+  server.run('nohup ipmi_sim -n > /var/log/ipmi_sim.log 2>&1 &', verbose: true, check_errors: true)
 end
 
 When(/^the server stops mocking an IPMI host$/) do
   get_target('server').run('pkill ipmi_sim')
-  get_target('server').run('pkill fake_ipmi_host.sh || :')
+  get_target('server').run("ps aux | grep [f]ake_ipmi_host.sh | awk '{print $2}' | xargs kill")
```

Comment:
I would stick with the previous version, there are two issues with it:
1. Certain implementations of `ps` might lead to a different output, thus it may require extra handling in the future.
2. `|| :`  is kind of a safety mechanism, if grep won't find this process it may lead to errors.
3.  It involves more moving components: `ps`, `grep`, `awk`, and `xargs`, `kill` whereas there you just have `pkill`.

Reply:
I agree in your points @szachovy but @maximenoel8 said that pkill was not killing successfully, and we need to address it somehow.

Reply:
If you look in uyuni podman you will see the pkill is not working: 
```hcl
uyuni-master-podman-srv:~ #  ps aux | grep fake_ipmi_host.sh
root     10582  0.0  0.0   4236  3120 ?        S    Oct16   0:03 /bin/bash /etc/ipmi/fake_ipmi_host.sh
root     11787  0.0  0.0   9156  2336 pts/0    S+   09:48   0:00 grep --color=auto fake_ipmi_host.sh
root     13152  0.0  0.0   4236  3104 ?        S    Oct16   0:03 /bin/bash /etc/ipmi/fake_ipmi_host.sh

```

Reply:
`--full` is missing:
```bash
uyuni-master-podman-srv:~ # ps aux | grep fake_ipmi_host.sh
root     10582  0.0  0.0   4236  3120 ?        S    Oct16   0:03 /bin/bash /etc/ipmi/fake_ipmi_host.sh
root     13152  0.0  0.0   4236  3104 ?        S    Oct16   0:03 /bin/bash /etc/ipmi/fake_ipmi_host.sh
root     13807  0.0  0.0   8780  2328 pts/1    S+   10:03   0:00 grep --color=auto fake_ipmi_host.sh
uyuni-master-podman-srv:~ # pkill --full fake_ipmi_host.sh
uyuni-master-podman-srv:~ # ps aux | grep fake_ipmi_host.sh
root     13877  0.0  0.0   8780   816 pts/1    S+   10:03   0:00 grep --color=auto fake_ipmi_host.sh
```

Reply:
Nice, it seems the proper fix for that killing, indeed.